arches/app/etl_modules/base_import_module.py

```diff
@@ -367,27 +370,147 @@ def start(self, request):
 
             return response
 
+    def cli(self, source):
+        initiated = self.start(self.request)
+
+        if initiated["success"]:
+            read = self.read_for_cli(source)
+        else:
+            return {
+                "success": False,
+                "data": {"title": _("Error"), "message": initiated["data"]["message"]},
+            }
+
+        if read["success"]:
+            self.request.POST.__setitem__(
+                "load_details", json.dumps({"result": read["data"]})
+            )
+            self.request.POST.__setitem__("multiprocessing", True)
+            written = self.write(self.request)
+        else:
+            return {
+                "success": False,
+                "data": {"title": _("Error"), "message": read["data"]["message"]},
+            }
+
+        if written["success"]:
+            return {"success": True, "data": "done"}
+        else:
+            return {
+                "success": False,
+                "data": {"title": _("Error"), "message": written["data"]["message"]},
+            }
+
+    def read_for_cli(self, source):
+        self.cumulative_files_size = 0
+        self.temp_dir = os.path.join(settings.UPLOADED_FILES_DIR, "tmp", self.loadid)
+        try:
+            self.delete_from_default_storage(self.temp_dir)
+        except FileNotFoundError:
+            pass
+        file_name = os.path.basename(source).split("/")[-1]
+        file_stat = os.stat(source)
+
+        result = {
+            "summary": {
+                "name": file_name,
+                "size": self.filesize_format(file_stat.st_size),
+                "files": {},
+            }
+        }
+        validator = FileValidator()
+        if len(validator.validate_file_type(source)) > 0:
+            return {
+                "status": 400,
+                "success": False,
+                "title": _("Invalid excel file/zip specified"),
+                "message": _("Upload a valid excel file"),
+            }
+
+        if source.split(".")[-1].lower() == "zip":
+            with zipfile.ZipFile(source, "r") as zip_ref:
+                files = zip_ref.infolist()
+                for file in files:
+                    if file.filename.split(".")[-1] == "xlsx":
+                        self.cumulative_files_size += file.file_size
+                    if not file.filename.startswith("__MACOSX"):
+                        if not file.is_dir():
+                            result["summary"]["files"][file.filename] = {
+                                "size": (self.filesize_format(file.file_size))
+                            }
+                            result["summary"][
+                                "cumulative_files_size"
+                            ] = self.cumulative_files_size
+                        default_storage.save(
+                            os.path.join(self.temp_dir, file.filename),
+                            File(zip_ref.open(file)),
+                        )
```

Comment:
Looks like this executes even if the file.is_dir() returns true. Is that wanted? Maybe the two if's for MACOSX and is_dir() can be combined?

Reply:
Also, we can make sure to use `with` to open the file so we don't leave it open?